main/src/ca/uwaterloo/flix/language/phase/LoadBytecode.scala

```diff
@@ -233,4 +320,67 @@ object LoadBytecode extends Phase[ExecutableAst.Root, ExecutableAst.Root] {
     }.toMap
   }
 
+  /**
+    * Find enums from the given expression
+    * @param e expression
+    * @return
+    */
+  def findEnums(e: Expression): List[(Type, (String, Type))] = e match {
```

Comment:
It should be possible to move this into EnumGen (at least after there is no dependencies on Class[_])